<?xml version="1.0"?>
<launch>
  <arg name="arm_1_joint_cmd_pos_name" default="/iiwa/effort_j1_position_controller/command" />
  <arg name="arm_2_joint_cmd_pos_name" default="/iiwa/effort_j2_position_controller/command" />
  <arg name="arm_3_joint_cmd_pos_name" default="/iiwa/effort_j4_position_controller/command" />
  <arg name="up_down_lower_limit" default="[-2.8]" />
  <arg name="up_down_upper_limit" default="[2.8]" />
  <arg name="left_right_lower_limit" default="[-2.8]" />
  <arg name="left_right_upper_limit" default="[2.8]" />
  <arg name="near_far_lower_limit" default="[-2.8, -2.8]" />
  <arg name="near_far_upper_limit" default="[2.8, 2.8]" />
  <arg name="joint_states_topic" default="/iiwa/joint_states" />
  <arg name="shoulder_frame" default="iiwa_link_0" />
  <arg name="tcp_frame" default="iiwa_link_6" />
  <arg name="joint_names" default="['iiwa_joint_1', 'iiwa_joint_2', 'iiwa_joint_3', 'iiwa_joint_4', 'iiwa_joint_5', 'iiwa_joint_6', 'iiwa_joint_7']" />
  <arg name="arm_3_joint_index" default="4" />
  <arg name="follow_joint_trajectory_param" default="/iiwa/effort_trajectory_controller/follow_joint_trajectory" />
  <arg name="next_pos_delta_factor" default="0.7" />
  <arg name="map_phi_to_2pi" default="False" />
  <arg name="pred_pos_frame" default="world" />
  <arg name="pred_pos_topic" default="/pred_pos" />
  <arg name="target_position_topic" default="/target_position" />

  <node name="pub_target_as_marker" pkg="target_reaching_nengo" type="pub_target_pos_as_vis_marker.py" output="screen"/>

  <node name="pub_pred_pos_as_target" pkg="target_reaching_nengo" type="pub_kuka_target_position.py" output="screen">
    <param name="pred_pos_frame" value="$(arg pred_pos_frame)" />
    <param name="pred_pos_topic" value="$(arg pred_pos_topic)" />
    <param name="target_position_topic" value="$(arg target_position_topic)" />
  </node>

  <node name="Main_TR_CL_nengo_node" pkg="target_reaching_nengo" type="main_TR_CL_node.py" output="screen">
    <param name="arm_1_joint_cmd_pos_name" value="$(arg arm_1_joint_cmd_pos_name)" />
    <param name="arm_2_joint_cmd_pos_name" value="$(arg arm_2_joint_cmd_pos_name)" />
    <param name="arm_3_joint_cmd_pos_name" value="$(arg arm_3_joint_cmd_pos_name)" />
    <rosparam param="up_down_lower_limit" subst_value="True">$(arg up_down_lower_limit)</rosparam>
    <rosparam param="up_down_upper_limit" subst_value="True">$(arg up_down_upper_limit)</rosparam>
    <rosparam param="left_right_lower_limit" subst_value="True">$(arg left_right_lower_limit)</rosparam>
    <rosparam param="left_right_upper_limit" subst_value="True">$(arg left_right_upper_limit)</rosparam>
    <rosparam param="near_far_lower_limit" subst_value="True">$(arg near_far_lower_limit)</rosparam>
    <rosparam param="near_far_upper_limit" subst_value="True">$(arg near_far_upper_limit)</rosparam>
    <param name="joint_states_topic" value="$(arg joint_states_topic)" />
    <param name="shoulder_frame" value="$(arg shoulder_frame)" />
    <param name="tcp_frame" value="$(arg tcp_frame)" />
    <rosparam param="joint_names" subst_value="True">$(arg joint_names)</rosparam>
    <param name="arm_3_joint_index" value="$(arg arm_3_joint_index)" />
    <param name="next_pos_delta_factor" value="$(arg next_pos_delta_factor)" />
    <param name="map_phi_to_2pi" value="$(arg map_phi_to_2pi)" />
  </node>
</launch>
